astropy/table/column.py

```diff
@@ -1230,7 +1237,9 @@ def __setitem__(self, index, value):
 
         # update indices
         self.info.adjust_indices(index, value, len(self))
-        ma.MaskedArray.__setitem__(self, index, value)
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore', MaskedArrayFutureWarning)
```

Comment:
The `if .. else ..` will be negligible compared to other stuff in the setitem.  So I would suggest to make the change and put in a `HAS_MASKED_ARRAY_FUTURE_WARNING` variable in the import try/except above.